plugins/uefi/fu-plugin-uefi.c

```diff
@@ -351,6 +351,7 @@ fu_plugin_coldplug (FuPlugin *plugin, GError **error)
 		fu_device_set_id (dev, id);
 		fu_device_add_guid (dev, guid);
 		fu_device_set_version (dev, version);
+		fu_device_set_vendor (dev, NULL); //FIXME?
```

Comment:
I'd say you can get this from smbios and use that

Reply:
I did want to discuss this hunk. I think you use the smbios data only if the type is `FWUP_RESOURCE_TYPE_SYSTEM_FIRMWARE` -- I have no idea what to use for `FWUP_RESOURCE_TYPE_DEVICE_FIRMWARE` or the others. There must be a way of getting the vendor ID or at least *something* from the iter.

Reply:
Yeah I think it's definitely right to do that when it's the system firmware type.  Actually that reminds me that the same check should be put around setting product name here: https://github.com/hughsie/fwupd/blob/master/plugins/uefi/fu-plugin-uefi.c#L311

As for device firmware vendor - ESRT is not prescriptive of this. See kernel ABI: https://www.kernel.org/doc/Documentation/ABI/testing/sysfs-firmware-efi-esrt

I think you should leave it unset for devices.  From one manufacturer to another it may be implemented as a system vendor distributed firmware (so is it a system vendor ID?) while others the IHV may distribute directly.  That's a level of detail you shouldn't be able to tell from fwupd.